### PR TITLE
doc

### PR DESCRIPTION
**Pull request by hsouf**
WIP

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,14 +102,26 @@ sudo apt install -y libgmp3-dev # linux
 brew install gmp # mac
 ```
```
Comment (achab) [inline README.md:105]: `nvm install 16` disappeared, but it should still be on the readme

Comment (hsouf) [reply README.md:105]: noted

File: README.md [hunk 2]
```diff
@@ -53,6 +171,12 @@ Then load all the environment variables
 ```bash
 source .evn/*
 ```
+To enable lite mode
```
Comment (swapnilraj) [inline README.md:174]: This bit isn't relevant

Comment (hsouf) [reply README.md]: ack

File: README.md [hunk 3]
```diff
@@ -1,18 +1,122 @@
...
+- The amount withdraw of ETHStaticAToken will be burned by the bridge
+- A message will be sent to L1 with the L1 token address, the l1 recipient, and the amount
+
+(@TODO: add more info on how we bridge the current rewards index when withdrawing..)
+ 
+
+
+## Synchronisation of rewards on L1 <> L2
+
+The challenge here was to allow users to continue enjoying the same rewards as on L1 by continously updating -whenever is possible- the `rewards_index` of all ETHStaticATokens to match the value of their respective StaticATokens on L1. To achieve that, we have updated the `_beforeBeforeTokenTransfer()` function on the new implementation of staticATokens (`StaticATokensLMNew.sol`) to send a message (through the bridge) with the latest `reward_index` value.
```
Comment (JorikSchellekens) [inline README.md:81]: I don't think our users need to know the design changes of aave's static tokens. Just how to use the system

File: README.md [hunk 4]
```diff
@@ -1,18 +1,128 @@
...
+
+ <a href="https://github.com/aave-starknet-project/aave-starknet-bridge/blob/main/LICENSE.md"><img src="https://img.shields.io/github/license/sourcerer-io/hall-of-fame.svg?colorB=ff0000"></a>
+
+:warning: This codebase is still in an experimental phase, has not been audited, might contain bugs and should not be used in production.
+
+## Overview
+
+For Aave, one of the main current and future goals is growth of liquidity and user base. As seen on side-chains with low transaction cost like Polygon or Avalanche, there is high demand to use the Aave protocol with small amounts to earn high yield.
+
+Both previous points show that having an initial phase on the Aave <> Starknet integration allowing deposit/withdrawal on Aave Ethereum by exclusively transacting on Starknet can be a good idea, target-wise.
```
Comment (JorikSchellekens) [inline README.md:16]: This needs to be updated since there are no longer two prior points

File: README.md [hunk 5]
```diff
@@ -1,18 +1,128 @@
...
+  *  `Proxy` - A proxy implementation 
+
+`L2`
+  * `ETHStaticAToken` - Tokens on Starknet equivalent to each staticAToken on Ethereum mainnet. Contains the same logic for tracking user rewards as staticATokens and has the same `rewards_index`.
+  * `claimable` - tracks users' claimable rewards and current reward index for each `ETHStaticAToken`
+  * `token_bridge` - is responsible for:
+    * bridging the staticATokens to and from L2. Minting and burning ETHStaticATokens on message from L1. 
+    * bridging rewAAVE token back to L1
+    * updating `rewards_index` for each ETHStaticAToken on message from L1 
+  * `rewAAVE` - a very basic ERC20 to represent the rewards on L2
```
Comment (JorikSchellekens) [inline README.md:40]: Maybe we should leave out 'a very basic' and instead just say:
'an ERC20 representing the rewards on L2'

File: README.md [hunk 6]
```diff
@@ -1,18 +1,128 @@
...
+### Transfer L2->L1:
+
+To bridge their staticATokens back to L1, users need to call `initiate_withdraw()` on Starknet `token_bridge`. 
+
+Calling `initiate_withdraw` will result in the following:
+
+- The amount withdraw of ETHStaticAToken will be burned by the bridge
+- A message will be sent to L1 with the L1 token address, the l1 recipient, and the amount
+
+(@TODO: add more info on how we bridge the current rewards index when withdrawing..)
```
Comment (JorikSchellekens) [inline README.md:82]: Just needs to read:

We track the reward index on departure of the static token and send the rewards accrued during the bridging process to the recipients address.

**Review by JorikSchellekens — APPROVED** [review]

File: README.md [hunk 7]
```diff
@@ -1,18 +1,120 @@
...
+  * `token_bridge` - is responsible for:
+    * bridging the staticATokens to and from L2. Minting and burning ETHStaticATokens on message from L1. 
+    * bridging rewAAVE token back to L1
+    * updating `rewards_index` for each ETHStaticAToken on message from L1 
+  * `rewAAVE` - an ERC20 representing the rewards on L2
+  *  `proxy` - a generic implementation of a proxy in starknet
+
+## ETHStaticATokens on L2
+
+Natively, on Aave aTokens grow in balance, not in value. To be able to create this kind of model, it is important to wrap them before bridging, converting them in a token that grows in value, not in balance.
```
Comment (achab) [inline README.md:43]: "on Aave aTokens" --> "Aave tokens"

File: README.md [hunk 8]
```diff
@@ -1,18 +1,120 @@
...
+
+ETHStaticATokens are an implementation of the wrapped aTokens that will continuously increase in value on Starknet because they are backed by the increasing staticATokens amounts locked in the bridge contract on Ethereum. The ETHStaticATokens can then be bridged back to staticATokens.
```
Comment (achab) [inline README.md:45]: "The ETHStaticATokens" --> "ETHStaticATokens"

File: README.md [hunk 9]
```diff
@@ -1,18 +1,120 @@
...
+Calling `deposit` will result in the following:
+
+- staticATokens will be transfered from the user account to the L1 bridge
+- A `deposit` event will be emitted with the L1 token address, the recipient address on L2, and the amount
+- A message will be sent to the  L2 bridge with the amount to be transferred, the L1 token address and the recipient address as parameters.
+- The token bridge on L2 will then be minting the correspending ETHStaticAToken of the L1 token to the user.
+
+Calling `depositUnderlying` will result in the following:
+
+Users will have to approve the bridge to spend the underlying `asset` tokens or `aTokens`, depending on the provided value for `fromAsset` argument when depositing.
```
Comment (achab) [inline README.md:62]: correct ! this information should be reflected elsewhere in the doc. the bridge can receive staticATokens (by calling deposit), aTokens (by calling depositUnderlying) or the underlying asset (by calling depositUnderlying).

File: README.md [hunk 10]
```diff
@@ -1,18 +1,120 @@
...
+## Bridging staticATokens from L1<>L2
+
+### Transfer L1->L2: 
+
+Users can either bridge their staticAToken to L2 by calling `deposit()` on `TokenBridge`, or deposit the underlying asset of the staticAToken directly by calling `depositUnderlying()`.
+
+Calling `deposit` will result in the following:
+
+- staticATokens will be transfered from the user account to the L1 bridge
+- A `deposit` event will be emitted with the L1 token address, the recipient address on L2, and the amount
```
Comment (achab) [inline README.md:56]: let's be consistent with the naming : "with L1 token address, L2 recipient address and the amount"

File: README.md [hunk 11]
```diff
@@ -1,18 +1,120 @@
...
+- A message will be sent to the  L2 bridge with the amount to be transferred, the L1 token address and the recipient address as parameters.
+- The token bridge on L2 will then be minting the correspending ETHStaticAToken of the L1 token to the user.
+
+Calling `depositUnderlying` will result in the following:
+
+Users will have to approve the bridge to spend the underlying `asset` tokens or `aTokens`, depending on the provided value for `fromAsset` argument when depositing.
+
+- the underlying asset of the staticAToken or the aToken (depending on the user's input )  will be transfered from the user account to the L1 bridge
```
Comment (achab) [inline README.md:64]: sorry for the format comment but sentences should start with a capital character and end with a dot.

File: README.md [hunk 12]
```diff
@@ -1,18 +1,120 @@
...
+- the bridge will then deposit the aTokens/asset on the staticAToken
+- new staticATokens will be minted to the bridge
+- A message will be sent to the  L2 bridge with the amount to be transferred, the L1 token address and the recipient address as parameters.
+- The token bridge on L2 will then be minting the correspending ETHStaticAToken of the L1 token to the user.
+
+
+### Transfer L2->L1:
+
+To bridge their staticATokens back to L1, users need to call `initiate_withdraw()` on Starknet `token_bridge`.
```
Comment (achab) [inline README.md:73]: "to call `initiate_withdraw` on Starknet"

File: README.md [hunk 13]
```diff
@@ -1,18 +1,120 @@
...
+
+- The amount to withdraw will be burned by the bridge
+- A message will be sent to L1 with the L1 token address, the l1 recipient, the l2 rewards index and the amount
+- The L1 bridge will then transfer the staticATokens to the l1 recipient
+- The L1 bridge also checks for any difference in the l1/l2 rewards index and transfers any unclaimed rewards to the l1 user
+
+
+## Synchronisation of rewards on L1 <> L2
+
+Starknet users will continue to enjoy the same rewards as on L1 after bridging their assets; To achieve that we continously update the `rewards_index` of all ETHStaticATokens to match the value of their respective StaticATokens on L1, by tracking the reward index on departure of the static token and sending the rewards accrued during the bridging process to the recipients address.
```
Comment (achab) [inline README.md:85]: "their assets;" --> "their assets."

File: README.md [hunk 14]
```diff
@@ -1,18 +1,120 @@
...
+
+
+## Claiming rewards on L2
+
+
+To claim rewards users need to call `claim_rewards()` on ETHStaticAToken contract. The ETHStaticAToken will then be calling the bridge to mint the due `rewAAVE` tokens to the user.
```
[the remark this answers is not in the document]
Comment (achab) [reply README.md:91]: "to call `claim_rewards`"

File: README.md [hunk 15]
```diff
@@ -1,18 +1,120 @@
...
+
+
+## Proxies
+
+All calls made to the following contracts will be handled by a proxy who delegates the calls to the available implementation of these contracts.
+- Token bridge on L2 
+- ETHStaticATokens 
+- Token bridge on L1 
+
+Using proxies during the early release of the bridge will allow us to upgrade and continuously improve the contracts implementations.
```
Comment (achab) [inline README.md:105]: This sentence is not needed

File: README.md [hunk 16]
```diff
@@ -1,18 +1,120 @@
...
+To claim rewards users need to call `claim_rewards()` on ETHStaticAToken contract. The ETHStaticAToken will then be calling the bridge to mint the due `rewAAVE` tokens to the user.
+
+## Bridging rewards from L2->L1
+
+To bridge their rewards to L1, users will have to call `bridge_rewards()` on the L2 `token_bridge` providing the `amount` to be bridged. The L1 bridge will be claiming pending rewards to self from all staticATokens in a loop until having enough rewards balance to transfer it to users. The claiming ans transferring of rewards is handled by `receiveRewards()` on `TokenBridge.sol`
+
+
+## Proxies
+
+All calls made to the following contracts will be handled by a proxy who delegates the calls to the available implementation of these contracts.
```
Comment (achab) [inline README.md:100]: I think other contracts should be added to your list. Almost all contracts are behind proxies now

**Comment by JorikSchellekens** [comment]
@hsouf I don't see massil's comments

File: README.md (hunk 15 above)
Comment (hsouf) [reply README.md:105]: agreed

**Pull request merged**